watcher/tool/watcher/main.go

```diff
@@ -18,33 +18,65 @@ package main
 
 import (
 	"context"
-	"flag"
 	"fmt"
 	"os"
 	"time"
 
 	"github.com/gravitational/monitoring-app/watcher/lib/constants"
+	"github.com/gravitational/monitoring-app/watcher/lib/influxdb"
 	"github.com/gravitational/monitoring-app/watcher/lib/kubernetes"
 
 	"github.com/gravitational/trace"
-	"github.com/gravitational/version"
 	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	flag "github.com/spf13/pflag"
 )
 
+var (
+	mode           string
+	debug          bool
+	influxDBConfig influxdb.Config
+
+	envs = map[string]string{
+		"INFLUXDB_ADMIN_USERNAME":    "influxdb-admin-username",
+		"INFLUXDB_ADMIN_PASSWORD":    "influxdb-admin-password",
+		"INFLUXDB_GRAFANA_USERNAME":  "influxdb-grafana-username",
+		"INFLUXDB_GRAFANA_PASSWORD":  "influxdb-grafana-password",
+		"INFLUXDB_TELEGRAF_USERNAME": "influxdb-telegraf-username",
+		"INFLUXDB_TELEGRAF_PASSWORD": "influxdb-telegraf-password",
+	}
+
+	rootCmd = &cobra.Command{
+		Use:          "watcher",
+		Short:        "Utility to manage InfluxDB/Grafana/Alerts",
+		SilenceUsage: true,
+		RunE:         root,
+	}
+)
+
+func init() {
+	rootCmd.PersistentFlags().StringVar(&mode, "mode", "", fmt.Sprintf("Watcher mode: %v", constants.AllModes))
+	rootCmd.PersistentFlags().StringVar(&influxDBConfig.InfluxDBAdminUser, "influxdb-admin-username", constants.InfluxDBAdminUser, "InfluxDB administrator username")
+	rootCmd.PersistentFlags().StringVar(&influxDBConfig.InfluxDBAdminPassword, "influxdb-admin-password", constants.InfluxDBAdminUser, "InfluxDB administrator password")
+	rootCmd.PersistentFlags().StringVar(&influxDBConfig.InfluxDBGrafanaUser, "influxdb-grafana-username", constants.InfluxDBGrafanaUser, "InfluxDB grafana username")
+	rootCmd.PersistentFlags().StringVar(&influxDBConfig.InfluxDBGrafanaPassword, "influxdb-grafana-password", constants.InfluxDBGrafanaUser, "InfluxDB grafana password")
+	rootCmd.PersistentFlags().StringVar(&influxDBConfig.InfluxDBTelegrafUser, "influxdb-telegraf-username", constants.InfluxDBTelegrafUser, "InfluxDB telegraf username")
+	rootCmd.PersistentFlags().StringVar(&influxDBConfig.InfluxDBTelegrafPassword, "influxdb-telegraf-password", constants.InfluxDBTelegrafUser, "InfluxDB telegraf password")
+
+	bindFlagEnv(rootCmd.PersistentFlags())
+}
+
 func main() {
-	var mode string
-	flag.StringVar(&mode, "mode", "", fmt.Sprintf("watcher mode: %v", constants.AllModes))
-	ver := flag.Bool("version", false, "print version")
-	flag.Parse()
-
-	if *ver {
-		version.Print()
-		return
+	if err := rootCmd.Execute(); err != nil {
+		log.Error(trace.DebugReport(err))
```

Comment:
Latest release allows to avoid trace escaping issues (which renders the trace unreadable) with:
```go
log.WithError(err).Warn("Command failed.") // Logs strack trace in readable form
```